journals/apps/api/filters.py

```diff
@@ -78,3 +80,31 @@ class Meta:
             'page_id',
             'last_visit'
         ]
+
+
+class PageAuthorizationFilter(BaseFilterBackend):
+    """
+    Filter that only allows user to see pages they have access to.
+    """
+    def filter_queryset(self, request, queryset, view):
+        # TODO: Currently has additional DB calls for gettings authorized journals and authorized pages attached to
+        # the journals. This happens on every API call
+        authorized_journal_ids = [
+            journal.id
+            for journal
+            in JournalAccess.get_user_accessible_journals(request.user)
+        ] if request.user.is_authenticated else []
```

Comment:
This loop is unnecessary and will be more efficient if you return a queryset from JournalAccess.get_user_accessible_journals() instead of list. Then you can do the following:
```
authorized_journal_ids = JournalAccess.get_user_accessible_journals(request.user).values_list("id", flat=True)

authorized_journal_page_ids = JournalPage.objects.filter(
            journal_about_page__journal__id__in=authorized_journal_ids
        ).values_list('id', flat=True)
```

Reply:
Fixed